src/workers/wallets/worker.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import utils from 'jwallet-web-keystore'
+
+import keystore from 'services/keystore'
+
+import {
+  CREATE_REQUEST,
+  CHECK_NAME_REQUEST,
+  createError,
+  createSuccess,
+  checkNameError,
+  checkNameSuccess,
+} from 'routes/Wallets/routes/Create/modules/walletsCreate'
+
+import type { WalletsCreateAction } from 'routes/Wallets/routes/Create/modules/walletsCreate'
+
+export type WalletsWorkerMessage = {|
+  +data: WalletsCreateAction,
+|}
+
+// eslint-disable-next-line fp/no-mutation
+onmessage = (msg: WalletsWorkerMessage): void => {
+  const action: WalletsCreateAction = msg.data
+
+  switch (action.type) {
+    case CHECK_NAME_REQUEST: {
+      try {
+        const { wallets, name } = action.payload
+
+        keystore.checkWalletUniqueness(wallets, name, 'name')
+
+        postMessage(checkNameSuccess())
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err)
```

Comment:
better to throw error

Reply:
it's inside worker
`console.error` - just to save stack trace and message in console
`throw` - will cause Uncaught Error, but actually it was caught